### PR TITLE
De-dup C# global usings

### PR DESCRIPTION
**Pull request by amcasey**
...so that we don't incorrectly look in each copy and report a false
ambiguity (causing an assert).

Bonus: switch ```this``` to ```originalBinder``` in a SemanticModel import
case.

Part of #5450.

### CODE REVIEW

**Comment by amcasey** [comment]
@AlekseyTs VB seems to be unaffected - it actually generates a diagnostic for the duplicate using and then suppresses the diagnostic.

**Comment by amcasey** [comment]
FYI @AlekseyTs @VSadov @gafter @jaredpar @agocke 
FYI @tmat @cston @KevinH-MS @genlu @ManishJayaswal

**Comment by amcasey** [comment]
Inheriting global usings from previous submissions will follow in a separate PR.

**Comment by amcasey** [comment]
Filed https://github.com/dotnet/roslyn/issues/5485 for the incomplete design of global using static.

File: src/Compilers/CSharp/Portable/Binder/Imports.cs [hunk 1]
```diff
@@ -221,15 +221,16 @@ public static Imports FromGlobalUsings(CSharpCompilation compilation)
             var diagnostics = new DiagnosticBag();
             var usingsBinder = new InContainerBinder(compilation.GlobalNamespace, new BuckStopsHereBinder(compilation));
             var boundUsings = ArrayBuilder<NamespaceOrTypeAndUsingDirective>.GetInstance();
+            var uniqueUsings = PooledHashSet<NamespaceOrTypeSymbol>.GetInstance();
 
-            foreach (string ns in usings)
+            foreach (string targetString in usings)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: Why 'target? Perhaps 'using' instead.

File: src/Compilers/CSharp/Portable/Binder/Imports.cs [hunk 2]
```diff
@@ -300,9 +307,14 @@ public static Imports FromGlobalUsings(CSharpCompilation compilation)
 
         private void MarkImportDirective(CSharpSyntaxNode directive, bool callerIsSemanticModel)
         {
-            if (directive != null && _compilation != null && !callerIsSemanticModel)
+            MarkImportDirective(_compilation, directive, callerIsSemanticModel);
+        }
+
+        private static void MarkImportDirective(CSharpCompilation compilation, CSharpSyntaxNode directive, bool callerIsSemanticModel)
+        {
+            if (directive != null && compilation != null && !callerIsSemanticModel)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: Is 'compilation' ever null?

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Binder/Imports.cs (hunk 1 above)
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: Done.

File: src/Compilers/CSharp/Portable/Binder/Imports.cs (hunk 2 above)
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: ```compilation``` is null in ```Imports.Empty```.

Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: Are we ever calling MarkImportDirective on Imports.Empty?

Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/Imports.cs]: I've converted the condition into an assert, but I'll merge it as part of a subsequent PR.

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**